files/uncompress.py

```diff
@@ -0,0 +1,265 @@
+    """
+    try:
+        f_out = open(dest, 'wb')
+        f_in = bz2.BZ2File(src, 'rb')
+        try:
+            shutil.copyfileobj(f_in, f_out)
+        finally:
+            f_out.close()
+            f_in.close()
+        msg=""
+    except Exception, e:
+        msg="%s" % e
+
+    return msg
+
+def unxzip(module, src, dest):
+    """
+    Uncompress xz files. Since we must support python 2.4 (EL 5) we cannot import lzma and use native python.
+    Also, xz does not support a destination flag, so we must use > to redirect stdout to file. Really lame.
+    """
+    cmd_path = module.get_bin_path('xz')
+    cmd = '%s -c -d %s > %s' % (cmd_path, src, dest)
+    try:
+        module.run_command(cmd, use_unsafe_shell=True)
+        msg = ""
+    except Exception, e:
+        msg="%s" % e
+
+    return msg
+
+def filetype(module, src):
+    """
+    Get the filetype from the unix command file, and then pick the correct compression method to use for it
+    """
+    cmd_path = module.get_bin_path('file')
+    cmd = "%s -b -i %s" % (cmd_path, src)
+    ftype = module.run_command(cmd, use_unsafe_shell=True)
+
+    return ftype
+
+def copyfile(src, dest, deep_check):
+    """
+    Copy file from tempsrc to final destination. Unless its already at dest, and the same as tempsrc.
+    """
+    changed = False
+    if os.path.isfile(dest):
+        # This takes a long time
+        if deep_check is True:
+            nodiff = filecmp.cmp(src, dest, shallow=True)
+        # This is much quicker
+        else:
+            destsize = os.path.getsize(dest)
+            srcsize  = os.path.getsize(src)
+            if destsize != srcsize:
+                nodiff = False
+            else:
+                nodiff = True
+        # If there is a difference, then we change the destination
+        if nodiff is False:
+            shutil.move(src, dest)
+            changed = True
+    # If the destination file does not exist, then place it.
+    else:
+        shutil.move(src, dest)
+        changed = True
+
+    return changed
+
+def main():
+    module = AnsibleModule(
+        # not checking because of daisy chain to file module
+        argument_spec = dict(
+            src               = dict(required=True),
+            dest              = dict(required=True),
+            copy              = dict(default=True, type='bool'),
+            original_basename = dict(required=False), # used to handle 'dest is a directory' via template, a slight hack
+            deep_check        = dict(default=False, type='bool'), # This check takes a long time if dest already exists.
+        ),
+        add_file_common_args=True,
+    )
+
+    src         = os.path.expanduser(module.params['src'])
+    dest        = os.path.expanduser(module.params['dest'])
+    copy        = module.params['copy']
+    deep_check  = module.params['deep_check']
+    file_args   = module.load_file_common_arguments(module.params)
+    tempdir     = os.path.dirname(__file__)
```

Comment:
Do we really want to make all download in the user home dir ? This might cause issue if people use it on NFS or have quota ?

But if we switch to /tmp, it should go in a specific tmp directory, and sometime /tmp can have quota too.